core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -1334,6 +1334,42 @@ public void testOptionalWithDefault() throws B2JsonException {
         }
     }
 
+    @Test
+    public void testVersionRangeBackwards() throws B2JsonException {
+        thrown.expectMessage("last version 1 is before first version 2");
+        bzJson.toJson(new VersionRangeBackwardsClass(5));
+    }
+
+    private static class VersionRangeBackwardsClass {
+        @B2Json.required
+        @B2Json.versionRange(firstVersion = 2, lastVersion = 1)
+        private final int n;
+
+        @B2Json.constructor(params =  "n")
+        private VersionRangeBackwardsClass(int n) {
+            this.n = n;
+        }
+    }
+
+    @Test
+    public void testConflictingVersions() throws B2JsonException {
+        thrown.expectMessage("must not specify both 'firstVersion' and 'versionRange'");
```

Comment:
it might be more dev-friendly if this message included the name of the class with the issue.

Reply:
done.

Reply:
ok